platform-workflow-module/src/main/java/com/softicar/platform/workflow/module/workflow/version/export/WorkflowVersionExporter.java

```diff
@@ -0,0 +1,40 @@
+package com.softicar.platform.workflow.module.workflow.version.export;
+
+import com.google.gson.GsonBuilder;
+import com.softicar.platform.common.core.exceptions.SofticarIOException;
+import com.softicar.platform.common.io.mime.MimeType;
+import com.softicar.platform.common.io.writer.ManagedWriter;
+import com.softicar.platform.common.io.writer.OutputStreamWriterFactory;
+import com.softicar.platform.common.string.charset.Charsets;
+import com.softicar.platform.dom.document.IDomDocument;
+import com.softicar.platform.workflow.module.workflow.export.WorkflowDtoV1Exporter;
+import com.softicar.platform.workflow.module.workflow.version.AGWorkflowVersion;
+import java.io.IOException;
+
+public class WorkflowVersionExporter {
+
+	private final AGWorkflowVersion workflowVersion;
+
+	public WorkflowVersionExporter(AGWorkflowVersion workflowVersion) {
+
+		this.workflowVersion = workflowVersion;
+	}
+
+	public void export(IDomDocument document) {
+
+		var workflowDto = new WorkflowDtoV1Exporter(workflowVersion).exportWorkflow();
+		var workflowJson = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create().toJson(workflowDto);
+
+		var export = document//
+			.getEngine()
+			.createExport()
+			.setFilename(workflowVersion.getWorkflow().getName() + ".json")
```

Comment:
`AGWorkflowVersion` should really have a version number column. And we should include that version number in this file name. Exported files would be easier to recognize/manage that way.

Reply:
I don't see how we should generate such versions, other than e.g. using the hash over exported file.

Reply:
Yea... hm, should we at least include a date? Because need a way to tell versions (and files) apart. To organize them, to handle them, etc.